connexion/api.py

```diff
@@ -84,12 +84,17 @@ def __init__(self, swagger_yaml_path, base_url=None, arguments=None,
                             'swagger_url': swagger_url,
                             'auth_all_paths': auth_all_paths})
         arguments = arguments or {}
-        with swagger_yaml_path.open() as swagger_yaml:
-            swagger_template = swagger_yaml.read()
+        with swagger_yaml_path.open(mode='rb') as swagger_yaml:
+            contents = swagger_yaml.read()
+            try:
+                swagger_template = contents.decode()
+            except UnicodeDecodeError:
+                swagger_template = contents.decode('utf-8', 'replace')
+
             swagger_string = jinja2.Template(swagger_template).render(**arguments)
             self.specification = yaml.load(swagger_string)  # type: dict
```

Comment:
Unrelated to this issue, but it would be better to use safe_load instead of load.